pkg/kubelet/kubelet_node_status.go

```diff
@@ -581,7 +581,7 @@ func (kl *Kubelet) tryUpdateNodeStatus(ctx context.Context, tryNumber int) error
 
 	node, changed := kl.updateNode(ctx, originalNode)
 	// no need to update the status yet
-	if !changed && !kl.isUpdateStatusPeriodExperid() {
+	if !changed && !kl.isUpdateStatusPeriodExpired() {
 		kl.markVolumesFromNode(node)
 		return nil
 	}
```

Comment:
if `updateNode` returns `changed==false`, meaning there is no change in what we want to report for the node status, why is it important to call patchNodeStatus once on node startup?

Reply:
I don't really understand the description of this PR - "there is a risk that the kubelets internal state (e.g. addresses and hostname) is never set until there is a change in the node's status" ... shouldn't `changed` be true if that is the case?

Reply:
Sorry, I have updated the description to clarify. The gist is that the kubelet service certificate CSR cannot be created until `getLastObservedNodeAddresses` returns an actual address, and that doesn't happen until `setLastObservedNodeAddresses` has been called, which doesn't happen anywhere else than `patchNodeStatus`.
Note that the current behavior is also a clear change from how it was before https://github.com/kubernetes/kubernetes/pull/128640 and that it doesn't respect `--node-status-update-frequency` until the node status has changed the first time. After that it will happily update it regularly, but before there is any change, it will just wait indefinitely.